### PR TITLE
Update loss file

### PR DESCRIPTION
**Pull request by borgesa**
With reference to issue #21:
I have done the update I described in the issue.

Now the loss functionality can use all PyTorch loss functions (by specifying the name in config.json). A template of a custom loss function class is included.

Remaining:
- Functionality that checks if config['loss'] matches one of the custom classes in 'model/loss.py'
    - When we implement this, we need to decide if custom loss functions ('model/loss.py' ones) or standard PyTorch losses has precedence (in case someone implements a loss function with the same name as PyTorch standard losses, either by accident or on purpose)

### CODE REVIEW

**Comment by SunQpark** [comment]
**for `loss.py` **
1. line 7 has minor typo: `*kwargs` need to be replaced with `**kwargs`
2. line 26: `super(_Loss, self)` need to be changed to `super(CustomLossClass, self)`
3. line 32: for the loss function and metrics I prefer using `(output, target)` over `(input, target)` since `input` is a reserved word in python. That way would also match my implementations of metric functions too.
**for other files**
It seems that you want to pass arguments to the pytorch loss class, than `train.py` and `config.json` should be changed too.
```python
"loss": "NLLLoss",
"loss_args": {
    reduction='elementwise_mean'
},
```
for config, and
```python 
loss = get_loss_function(config['loss'], **config['loss_args'])
```
for train.py (line23)

**Comment by SunQpark** [comment]
Thank you for quick response! 
It seems that there are some [discussions](https://discuss.pytorch.org/t/why-is-input-used-as-var-name-in-examples-and-pytorch-convention/185) about using the `input` as variable name. I was surprised that this comes first in the google search, although the keyword was 'pytorch convention' which does not  include 'input'
Anyway, I will merge this PR now, thank you.
